libs/pika/version/include/pika/version.hpp

```diff
@@ -12,53 +12,42 @@
 
 #include <cstdint>
 #include <string>
+#include <string_view>
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace pika {
     // Returns the major pika version.
-    PIKA_EXPORT std::uint8_t major_version();
+    PIKA_EXPORT constexpr std::uint8_t major_version();
```

Comment:
These have to be defined inline. I suggest adding a small unit test that confirms that they can actually be used in constant expressions.